app/tests/dojo_tests.py

```diff
@@ -0,0 +1,215 @@
+
+    def test_print_room_with_wrong_room_name(self):
+        """
+        test the print_room function if appropriate message
+        will be returned if a non existing room name is passed in as
+        value
+        """
+        Dojo.print_room("orange")
+        output = sys.stdout.getvalue().strip()
+        self.assertEqual(output, "The room with the name {} does not exist"
+                         .format("orange"))
+
+    def test_print_room(self):
+        """
+        test the print_room function if no warning message
+        will be returned if an existing room name is passed in as
+        value
+        """
+        new_office = Dojo.create_room("office", ["orange"])[0]
+        Dojo.print_room("orange")
+        output = sys.stdout.getvalue().strip()
+        self.assertNotEqual(output, "The room with the name {} does not exist"
+                            .format("orange"))
+
+    def test_print_allocations_with_existing_filename(self):
+        """
+        test the print_allocations for a warning
+        message if an already existing filename is
+        inputed as value.
+        """
+        filename = "file1.txt"
+        open(self.filepath + filename, "w").close()
+        Dojo.print_allocations("file1")
+        output = sys.stdout.getvalue().strip()
+        self.assertEqual(output, "file1.txt already exist")
+
+    def test_print_allocations_with_new_filename(self):
+        """
+        test the print_allocations whether it successfully create
+        a new file if a non-existing filename is passed as value.
+        """
+        filename = "file2.txt"
+        Dojo.print_allocations("file2")
+        self.assertTrue(path.isfile(self.filepath + filename))
+
+    def test_print_unallocated_with_existing_filename(self):
+        """
+        test the print_unallocated for a warning
+        message if an already existing filename is
+        inputed as value.
+        """
+        filename = "file3.txt"
+        open(self.filepath + filename, "w").close()
+        Dojo.print_unallocated("file3")
+        output = sys.stdout.getvalue().strip()
+        self.assertEqual(output, "file3.txt already exist")
+
+    def test_print_unallocated_with_new_filename(self):
+        """
+        test the print_unallocated whether it successfully create
+        a new file if a non-existing filename is passed as value.
+        """
+        filename = "file4.txt"
+        Dojo.print_allocations("file4")
+        self.assertTrue(path.isfile(self.filepath + filename))
+
+    def test_reallocate_person(self):
+        """
+        tests the reallocate_person function if it successfully
+        reallocates a person to a specified room
+        """
+        new_office = Dojo.create_room("office", ["red"])[0]
+        Dojo.add_person("koya", "gabriel", "staff")
+        person = Staff.get_staff_list()[0]
+        self.assertEqual(len(new_office.get_room_members()), 1)
```

Comment:
before we can assert this maybe after L204 we should assert that the room is empty..